socs/Lakeshore/Lakeshore336.py

```diff
@@ -0,0 +1,1195 @@
+        return self.sensor_type
+
+    def set_sensor_type(self, type):
+        '''Set the sensor type on the channel
+
+        Parameters
+        ----------
+        type : str
+            Sensor type must be in 'Disabled', 'Diode', 'Platinum RTD',
+            'NTC RTD', 'Thermocouple', 'Capacitance'
+        '''
+        assert type.lower() in sensor_lock
+
+        resp = self.get_input_type()
+        resp[1] = sensor_lock[type.lower()]
+        self.sensor_type = type.lower()
+        return self._set_input_type(resp)
+
+    def get_units(self):
+        '''Get the channel preferred units as plain text'''
+        self.get_input_type()
+        return self.units
+
+    def set_units(self, units):
+        '''Set the channel preferred units
+
+        Parameters
+        ----------
+        unit : str
+            Channel preferred units must be in 'Kelvin', 'Celsius, 'Sensor
+        '''
+        assert units.lower() in units_lock
+
+        resp = self.get_input_type()
+        resp[4] = units_lock[units.lower()]
+        self.units = units.lower()
+        return self._set_input_type(resp)
+
+    def get_input_name(self):
+        '''Get the name of the channel shown on front display
+
+        Returns
+        -------
+        str
+            The channel name for display purposes
+        '''
+        resp = self.ls.msg(f'INNAME? {self.input}').strip()
+        self.input_name = resp
+        return self.input_name
+
+    def set_input_name(self, name):
+        '''Set the name of the channel shown on front display
+
+        Parameters
+        ----------
+        name : str
+            The channel name for display purposes. Only send the first 15 characters
+        '''
+        name = name[:15]
+        self.input_name = name
+        resp = self.ls.msg(f'INNAME {self.input},{name}')
+        return resp
+
+    def get_input_curve(self):
+        '''Return the curve number of the curve assigned to this channel'''
+        resp = self.ls.msg(f'INCRV? {self.input}').strip()
+        self.input_curve = int(resp)
+        return self.input_curve
+
+    def set_input_curve(self, curve_num):
+        '''Set the curve number of the curve assigned to this channel'''
+        assert curve_num in range(1, 60)
+
+        self.input_curve = curve_num
+        resp = self.ls.msg(f'INCRV {self.input},{self.input_curve}')
+        return resp
+
+    def get_T_limit(self):
+        '''Return the temperature limit above which control outputs assigned this channel
+        shut off'''
+        resp = self.ls.msg(f'TLIMIT? {self.input}')
+        self.T_limit = float(resp)
+        return self.T_limit
+
+    def set_T_limit(self, limit):
+        '''Set the temperature limit above which control outputs assigned this channel
+        shut off'''
+        self.T_limit = limit
+        resp = self.ls.msg(f'TLIMIT {self.input},{self.T_limit}')
+        return resp
+
+# Curve class copied from socs/Lakeshore372.py on 2020/08/21
+class Curve:
```

Comment:
This looks largely unmodified. If this can be the same class as in the 372 you should import it from the 372 driver. You could use that 372 Curve as a base class and extend it with the one added method here.

Reply:
Compared to the Curve class in Lakeshore372.py, it looks like there are some small differences besides the addition of the new soft_cal() function. Functions like _set_header, set_format (only in the docstring), and _set_data_point differ due to different parameters in CRVHDR and CRVPT between the 372 and 336. set_curve() has an extra 1 second delay added after the curve is uploaded with the comment "necessary to make work." The get_header() function call within delete_curve() is commented out here but isn't in the 372 code.

Reply:
Thanks for the info, going to mark as resolved, since we're going to leave this for now.